packages/react/src/components/popover/popover.tsx

```diff
@@ -68,13 +71,16 @@ function PopoverWithPortal({
   ...props
 }: PopoverProps &
   Required<Pick<PopoverProps, 'align' | 'position' | 'target'>>) {
+  const domBody = useRef(document.body);
   const popover = useRef<HTMLDivElement>(null);
   const [placement, setPlacement] = useState([position, align] as const);
   const [anchor, setAnchor] = useState(at(target));
 
   useOnClickOutside(onClose, [target, popover]);
 
-  useObserver(
+  useResizeObserver(() => setAnchor(at(target)), [domBody]);
```

Comment:
This is the main fix line, the rest is implementing `useResizeObserver` and refactor/cleanup.